src/pages/QuizUi/CheckRadioRender.jsx

```diff
@@ -8,60 +8,84 @@ export default function checkRadioRender(
   questionType,
   selected,
   setSelected,
-  isCurrentQuestionAnswered
+  isCurrentQuestionAnswered,
+  correctnessInfo,
+  setSelectedOption
 ) {
   const checkbox = 'check-box';
   const handleOptionClick = (option) => {
     if (!isCurrentQuestionAnswered) {
       if (questionType === checkbox) {
-        setSelected((prevSelected) =>
-          prevSelected.includes(option)
-            ? prevSelected.filter((item) => item !== option)
-            : [...prevSelected, option]
-        );
+        const newSelected = selected.includes(option)
+          ? selected.filter((item) => item !== option) // Remove the option if it's already selected
+          : [...selected, option]; // Add the option if it's not already selected
+        setSelected(newSelected);
+        setSelectedOption(newSelected);
       } else {
+        // Radio button logic: set `selected` to the clicked option
         setSelected(option);
+        setSelectedOption(option); // Update the state in the parent component
       }
     }
   };
 
-  return options.map((option, index) => (
-    <Grid item xs={12} key={index}>
-      <Box
-        component="div"
-        onClick={() => handleOptionClick(option)}
-        sx={{
-          pl: 2,
-          border: '2px solid rgb(223, 221, 221)',
-          width: '100%',
-          borderRadius: 1,
-          boxSizing: 'border-box',
-          cursor: 'pointer', // Add a pointer cursor to indicate it's clickable
-        }}
-      >
-        <FormControlLabel
-          control={
-            questionType === checkbox ? (
-              <Checkbox
-                checked={selected.includes(option)}
-                value={option}
-                disabled={isCurrentQuestionAnswered}
-              />
-            ) : (
-              <Radio
-                checked={selected === option}
-                value={option}
-                disabled={isCurrentQuestionAnswered}
-              />
-            )
-          }
-          label={option}
+  return options.map((option, index) => {
+    const isOptionSelected =
+      questionType === checkbox
+        ? selected.includes(option)
+        : selected === option;
+
+    const borderStyle =
+      correctnessInfo && isOptionSelected
+        ? correctnessInfo.isCorrect
+          ? '2px solid green'
+          : '2px solid red'
+        : '2px solid rgb(223, 221, 221)';
+
+    return (
+      <Grid item xs={12} key={index}>
+        <Box
+          component="div"
+          onClick={() => handleOptionClick(option)}
           sx={{
-            width: '100%', // Ensure label takes the full width of the box
-            userSelect: 'none', // Prevent text selection on double click
+            pl: 2,
+            border: borderStyle,
+            width: '100%',
+            borderRadius: 1,
+            boxSizing: 'border-box',
+            cursor: 'pointer',
+            backgroundColor:
+              correctnessInfo && isOptionSelected
```

Comment:
I appreciate the feedback in the quiz immediately whether your question was correct!

I find the highlighted answer a bit difficult to read however. The contrast ratio between the red and the answer text is only 1.29. In general text at regular size should have a contrast with the background of 4 / 1. You can read more about what I mean here https://www.accessibility-developer-guide.com/knowledge/colours-and-contrast/text/. This is just for your background though! We don't need to be strict about the color contrast ratio here.

I'd suggest either making the background darker and the text close to white, or keeping the text darker and making the background lighter.

What do you think?

Reply:
Totally. Since we are using MUI components, the default "disabled" style of the radio group which automatically applies when I disable the UI after submitting an answer does make things hard to read.  I'll address this and make the change!